specs/CardanoShellSpec.lhs

```diff
@@ -0,0 +1,432 @@
+%format AND     = " && "
+%format WITHGUARDED = " && guarded"
+%endif
+
+% The following is the module header and some basic definitions
+% needed for the demo, to be included only in the code, but not
+% currently in the LaTeX output.
+%
+% Very few of the definitions are also added to help the formatting,
+% such as the Pair type and pattern synonym.
+
+%if style == newcode
+
+> {-# LANGUAGE PatternSynonyms #-}
+> {-# LANGUAGE StandaloneDeriving #-}
+> {-# OPTIONS_GHC -Wno-missing-methods #-}
+> module CardanoShellSpec where
+>
+> import Data.Map as Map
+> import Data.Set as Set
+>
+> type List a = [a]
+> type Pair a b = (a, b)
+> pattern Pair x y = (x, y)
+
+> implies :: Bool -> Bool -> Bool
+> x `implies` y = not x || y
+> infixr 1 `implies`
+
+> data ServerMessage 
+>    = QueryPort
+>    | Ping
+
+> instance Eq ServerMessage
+> instance Ord ServerMessage
+
+> data ClientMessage 
+>    = Started
+>    | Pong
+>    | ReplyPort Word16
+
+> instance Eq ClientMessage
+> instance Ord ClientMessage
+
+%endif
+
+% The following are some more specific formatting directives.
+% The spec code currently formats top-level functions differently
+% from local names / arguments, so we have formatting directives
+% for each of these top-level functions, which only apply when
+% producing LaTeX output.
+
+%if style /= newcode
+%format fullProtocol = "\named{fullProtocol}"
+%endif
+
+\begin{document}
+
+\title{Specification of the Cardano shell}
+
+\author{Kristijan Šarić}
+
+\date{06.02.2019}
+
+\maketitle
+
+\begin{abstract}
+This document is a high-level specification of how the pieces that make Cardano shell work together and an attempt to define a general, secure and simple way to combine them, while introducing the reader with the specifics of how they work.
+\textit{Beware, work in progress!}
+\end{abstract}
+
+\tableofcontents
+\listoffigures
+\listoftables
+
+\newpage
+\section{Introduction}
+\label{sec:introduction}
+
+The introduction. WIP!
+
+\newpage 
+\section{IPC and communication between Daedalus and Node}
+\label{sec:ipc}
+
+Currently, the \textit{Daedalus} and the \textit{Node} (this is what I'm calling the node, thus the uppercase) communicate via \textbf{IPC for reaching a consensus which port to use for the JSON API communication, since we have a lot of issues of ports being used} .
+\textbf{IPC} (Inter Process Communication) is a set of methods which processes can use to communicate - \url{https://en.wikipedia.org/wiki/Inter-process_communication}.\\
+
+The actual communication right now is being done by the \textit{spawn} function, pieces of which can be found \href{https://github.com/nodejs/node/blob/62942e9ad7a59b76e9255ea2560bad2245709efc/lib/internal/child_process.js#L306}{here}.
+The part of the code which adds the handle id which they will use to communicate via environment variable "NODE_CHANNEL_FD" \href{https://github.com/nodejs/node/blob/master/lib/internal/child_process.js#L324-L335}{here}.\\
```

Comment:
This section looks weird on my PDF reader.
![screenshot from 2019-02-12 14-17-36](https://user-images.githubusercontent.com/15665039/52613394-0812c080-2ed1-11e9-9b85-198f5e2a7821.png)

Reply:
Thanks, fixed.